pkg/http/api.go

```diff
@@ -249,6 +250,17 @@ func (a *api) constructMetadataEndpoints() []Endpoint {
 	}
 }
 
+func (a *api) constructHealthzEndpoints() []Endpoint {
```

Comment:
Is it for the user app to call into or for K8s? If its to be used by user app also, it should be exposed over gRPC as well. a user using gRPC clients to Dapr will have to create htp client and make http call for this. Also C#, Java SDK uses gRPC, users/SDK will need to make http clients to make this call when they are dealing only with gRPC,

Reply:
> Is it for the user app to call into or for K8s? If its to be used by user app also, it should be exposed over gRPC as well. a user using gRPC clients to Dapr will have to create htp client and make http call for this. Also C#, Java SDK uses gRPC, users/SDK will need to make http clients to make this call when they are dealing only with gRPC,

Had a discussion with @yaron2  and suggestion is to keep only http.  cc @yaron2

Reply:
If its meant for K8s only then http is fine. If its meant for user apps also then it should have gRPC also. A user app using only gRPC(for all the other calls) should not be forced to make this one call over http.

Reply:
As far as I understand, it is for K8s. unfortunately, k8s doesn't support gRPC natively. 

We may need to use the below tool. http endpoint is pretty cheap. maybe this endpoint can check grpc endpoint instead of using additional tool.. 
https://github.com/grpc-ecosystem/grpc-health-probe

Reply:
> As far as I understand, it is for K8s. unfortunately, k8s doesn't support gRPC natively.
> 
> We may need to use the below tool. http endpoint is pretty cheap. maybe this endpoint can check grpc endpoint instead of using additional tool..
> https://github.com/grpc-ecosystem/grpc-health-probe

That's true that K8s only support http but question is should we have gRPC endpoint for user app to call directly.

Reply:
This is strictly for the Kubelet to notify the Kube API server if the Pod should be removed from the traffic ring